### ISSUE TITLE
DatePicker's clear button is not accessible

### ISSUE THREAD

**Issue by mt9eet** [ant-design/ant-design#30496]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/date-picker](https://ant.design/components/date-picker)

### Steps to reproduce
- Select a date or time in the demo in order for the clear button to appear
- Open the browser inspection and check the DOM

### What is expected?
The clear button should be an HTML `button` element.

### What is actually happening?
The clear button is rendered as a HTML `span`, without a proper role to indicate that this is an interactive element.

| Environment | Info |
|---|---|
| antd | undefined |
| React | 16 |
| System | MacOS |
| Browser | Chrome |

---
This is not an Ant Design component issue, but an issue in the `rc-picker` component that it is built upon.

See https://github.com/react-component/picker/blob/master/src/Picker.tsx#L424

**Comment by mellis481** [comment]
@kerm1it I know it's two years old, but adding `role="button"` to a span does not solve the keyboard accessibility issue described in this ticket. I'm not sure why you thought it did or why @afc163 agreed that it did and merged your PR [#250](https://github.com/react-component/picker/pull/250). 

Additionally (@afc163), the current UX for clearing a DatePicker instance is horrible. The clear element is not only not keyboard accessible, all users will not know there is a clear button unless they hover over the calendar icon (which turns into the clear button on hover).

**Comment by afc163** [comment]
@mellis481 Any suggestion to fix it?

**Comment by mellis481** [comment]
@afc163 The clear button should be keyboard accessible and it should be visible without having to hover.
